### ISSUE TITLE
typedef before struct

### ISSUE THREAD

**Issue by alejandro-colomar** [include-what-you-use/include-what-you-use#1065]
I got this bug from iwyu(1) while running it on some NGINX code.  Okay, the code is not ideal, but it's correct.  I simplified it to a minimial reproducer:
```c
$ cat typedef.c
typedef struct foo_s foo_t;

struct foo_s {
    int x;
};
```
```
$ iwyu -Xiwyu --no_fwd_decls typedef.c

typedef.c should add these lines:
struct foo_s;

typedef.c should remove these lines:

The full include-list for typedef.c:
struct foo_s;
---
```

I kind of get the reason iwyu(1) suggests the fwd decl _even_ if I said I don't want them: if it suggested me to include the file that provides the declaration, it would be asking me to do a self-include, and of course it doesn't. 

In the real code, I found that there's a cyclic dep between typedefs, and that's why the typedef was before the struct definition.  A struct declaration before the typedef seems to me useless, so iwyu(1) should avoid warning on this if possible.

**Comment by kimgr** [comment]
Duplicates #793, but the fixing commit mentions this issue, so I'll close the other one.

**Comment by kimgr** [comment]
Bugfix merged with #1077
